spec/gist_spec.rb

```diff
@@ -1,17 +1,175 @@
-require_relative "../lib/gist"
+require 'fileutils'
+require 'webmock/rspec'
+require_relative '../lib/request'
+require_relative '../lib/gist'
 
 describe Gist do
+  gist = nil
+  gistContent = nil
+
+  before do
+    FileUtils.mkdir_p('dir_deep')
+    File.write('dir_deep/file.rb', 'file content')
+    FileUtils.mkdir_p('dir_deep/dir2')
+    File.write('dir_deep/dir2/file1.rb', 'file content')
+    FileUtils.mkdir_p('dir_shallow')
+    FileUtils.mkdir_p('dir_empty')
+    File.write('dir_shallow/file.rb', 'file content')
+    File.write('file.rb', 'file content')
+    File.write('empty.rb', '')
+
+  end
+
+  after do
+    File.delete('file.rb')
+    File.delete('empty.rb')
+    FileUtils.rm_r ('dir_deep')
+    FileUtils.rm_r ('dir_shallow')
+    FileUtils.rm_r ('dir_empty')
+  end
 
   describe 'new' do
+
     context 'When a file is given' do
+
+      before do
+        gist = Gist.new('file.rb', true, 'description')
+        gistContent = {
+          files: {
+            'file.rb' => {
+              content: 'file content'
+              }
+          },
+          description: 'description',
+          public: true
+        }
+      end
+
       it 'creates a new gist' do
-        expect(FileReader.read('file.rb')).to eql('file content')
+        expect(gist.gist).to eql(gistContent)
       end
     end
 
-    context 'When an empty file is given' do
-      it 'returns an empty string' do
-        expect(FileReader.read('empty.rb')).to eql('')
+    context 'When a shallow directory is given' do
+
+      before do
+        gist = Gist.new('dir_shallow', true, 'description')
+        gistContent = {
+          files: {
+            'file.rb' => {
+              content: 'file content'
+              }
+          },
+          description: 'description',
+          public: true
+        }
+      end
+
+      it 'creates a new gist' do
+        expect(gist.gist).to eql(gistContent)
+      end
+    end
+
+    context 'When a deep directory is given' do
+
+      before do
+        gist = Gist.new('dir_deep', true, 'description')
+        gistContent = {
+          files: {
+            'file.rb' => {
+              content: 'file content'
+            },
+            'file1.rb' => {
+              content: 'file content'
+            }
+          },
+          description: 'description',
+          public: true
+        }
+      end
+
+      it 'creates a new gist' do
+        expect(gist.gist).to eql(gistContent)
+      end
+    end
+
+    context 'When a empty directory is given' do
+
+      before do
+        gist = Gist.new('dir_empty', true, 'description')
+        gistContent = {
+          files: {},
+          description: 'description',
+          public: true
+        }
+      end
+
+      it 'creates a new gist' do
```

Comment:
in this tests that only have one line, try to use the online line `it`
https://relishapp.com/rspec/rspec-core/v/3-0/docs/subject/one-liner-syntax
(you have to define the subject)